src/librustc_trans/back/link.rs

```diff
@@ -190,6 +192,52 @@ pub(crate) fn link_binary(sess: &Session,
     out_filenames
 }
 
+/// Returns a boolean indicating whether we should preserve the object files on
+/// the filesystem for their debug information. This is often useful with
+/// split-dwarf like schemes.
+fn preserve_objects_for_their_debuginfo(sess: &Session) -> bool {
```

Comment:
I'm sort of hoping that we can use a function like this to enable split dwarf on Linux over time as well!